app/controllers/api/generic_object_definitions_controller.rb

```diff
@@ -104,8 +104,8 @@ def self.allowed_association_types
     end
 
     def self.allowed_types
-      GenericObjectDefinition::TYPE_MAP.keys.collect do |type|
-        [Dictionary.gettext(type.to_s, :type => :data_type, :notfound => :titleize, :plural => false), type.to_s]
+      GenericObjectDefinition::TYPE_NAMES.each.collect do |key, value|
```

Comment:
Does `GenericObjectDefinition::TYPE_NAMES.stringify_keys.invert.to_a.sort` suffice here?

Was this constant extracted for this use only? These transformations seem to suggest that its data structure is a poor fit 😉

Reply:
Yes, right now this is the only place where `enericObjectDefinition::TYPE_NAMES` is being used.

What you're suggesting definitely looks better.